tracing/opentracing/grpc.go

```diff
@@ -0,0 +1,56 @@
+package opentracing
+
+import (
+	"github.com/opentracing/opentracing-go"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
+)
+
+// ToGRPCRequest returns a grpc RequestFunc that injects an OpenTracing Span
+// found in `ctx` into the grpc Metadata. If no such Span can be found, the
+// RequestFunc is a noop.
+func ToGRPCRequest(tracer opentracing.Tracer) func(ctx context.Context, md *metadata.MD) context.Context {
+	return func(ctx context.Context, md *metadata.MD) context.Context {
+		if span := opentracing.SpanFromContext(ctx); span != nil {
+			// There's nothing we can do with an error here.
+			_ = tracer.Inject(span, opentracing.TextMap, metadataReaderWriter{md})
+		}
+		return ctx
+	}
+}
```

Comment:
Got it. Yeah I'm concerned because the strength of tracing relies so much on standardization of mechanism i.e. the way you shuttle the IDs between process boundaries... I'm so hungry for a spec here :)